ipaserver/install/custodiainstance.py

```diff
@@ -64,10 +65,22 @@ def create_instance(self):
                                                       realm=self.realm)
         sysupgrade.set_upgrade_state('custodia', 'installed', True)
 
+    def uninstall(self):
+        keystore = IPAKEMKeys({
+            'server_keys': self.server_keys,
+            'ldap_uri': self.ldap_uri
+        })
+        keystore.remove_server_keys()
+        installutils.remove_file(self.config_file)
+        sysupgrade.set_upgrade_state('custodia', 'installed', False)
```

Comment:
Haven't tested yet but won't this potentially leave the service still enabled post-install?

Reply:
How do I make sure the service is disabled?

Reply:
My concern was removing the upgrade state but in looking further I think that is ok. I do think you should move the call to super(CustodiaInstance, self).uninstall() first so that the service gets stopped before you remove the keys.

Reply:
Fixed